pkg/model/deployment_chain.proto

```diff
@@ -52,9 +52,32 @@ message ChainNode {
     ChainDeploymentRef deployment_ref = 2;
 }
 
+// ChainBlockStatus represents the current status of a block in deployment chain.
+enum ChainBlockStatus {
+    // DEPLOYMENT_BLOCK_UNKNOWN means the block is at unexpected status.
+    DEPLOYMENT_BLOCK_UNKNOWN = 0;
+    // DEPLOYMENT_BLOCK_PENDING means the block's nodes just initialized,
+    // all block nodes do not have its deploymentRef created.
+    DEPLOYMENT_BLOCK_PENDING = 1;
+    // DEPLOYMENT_BLOCK_RUNNING means at least one of the block's nodes has
+    // its deploymentRef and started its deployment flow.
+    DEPLOYMENT_BLOCK_RUNNING = 2;
+    // DEPLOYMENT_BLOCK_SUCCESS means all nodes in the block had finished with
+    // DEPLOYMENT_SUCCESS status.
+    DEPLOYMENT_BLOCK_SUCCESS = 3;
+    // DEPLOYMENT_BLOCK_FAILURE means one of the block's nodes had finished with
+    // DEPLOYMENT_FAILURE status.
+    DEPLOYMENT_BLOCK_FAILURE = 4;
+    // DEPLOYMENT_BLOCK_CANCELLED means one of the block's nodes had finished with
+    // DEPLOYMENT_CANCELLED status.
+    DEPLOYMENT_BLOCK_CANCELLED = 5;
+}
+
 message ChainBlock {
     // List of applications which should be deployed at the same time in chain.
     repeated ChainNode nodes = 1;
+    // Status of this deployment chain block.
+    ChainBlockStatus status = 2 [(validate.rules).enum.defined_only = true];
```

Comment:
Can you explain why we need this BlockStatus and where will we use this for?
I think the status of a block doesn't matter, we should only care about `chain` and `node` instead of `block`.

Reply:
Nice question! The `node` status is the status of its deployment and we already have `deploymentRef.Status` so we don't need that. About the `chain` status, we can determine based on this `block` status and currently we don't have any logic which uses the `chain` status. On the opposite, we use this `block` status on update node stub and determine the behavior of nodes in that block. Basically, once a block is completed, no nodes of the block are able to be updated (all will be locked) and I also use that block status to determine should a deployment of a node in block be processed (ref: [InChainDeploymentPlannable code](https://github.com/pipe-cd/pipe/pull/2866/files#diff-dc8d28c3b055fcad0f022406b744ce756e3be8c630b6c40aef12e026c6a58dcfR1214))